docs/advanced/pycpp/object.rst

```diff
@@ -168,3 +173,23 @@ Generalized unpacking according to PEP448_ is also supported:
     Python functions from C++, including keywords arguments and unpacking.
 
 .. _PEP448: https://www.python.org/dev/peps/pep-0448/
+
+.. _pytypes_gotchas:
+
+Gotchas
+=======
+
+Default-Constructed Wrappers
+----------------------------
+
+When a wrapper type is default-constructed, it is **not** a valid Python object (i.e. it is not ``py::none()``). It is simply the same as
+``(PyObject*)nullptr``. To check for this, either use ``(bool)my_wrapper`` or
```

Comment:
Minor detail, but can we get rid of the C casts?

E.g., "It is simply the same as a ``PyObject*`` null pointer", and "To check for this, use the (implicit) conversion of the wrapper object to ``bool``: ``static_cast<bool>(my_wrapper)``." (Do we need to mention the `ptr()` way, if there's a nice C++ operator?)

Reply:
I second avoiding C-style casts.

Reply:
Done. Good points!